src/app/rpmostree-builtin-upgrade.cxx

```diff
@@ -129,6 +164,51 @@ rpmostree_builtin_upgrade (int             argc,
         g_print ("note: automatic updates (%s) are enabled\n", policy);
     }
 
+  if (!opt_bypass_driver)
+    {
+      g_autofree char *update_driver_sd_unit = NULL;
+      g_autofree char *update_driver_name = NULL;
+      if (!get_driver_info (&update_driver_name, &update_driver_sd_unit, error))
+        return FALSE;
+
+      /* Notify user that an updates driver is registered and upgrades should be
+       * done through the driver. */
+      if (update_driver_sd_unit && update_driver_name)
+        {
+          g_print ("Error: Updates are driven by %s (%s)", update_driver_name, update_driver_sd_unit);
+
+          /* only try to get unit's `Documentation` if we're on the system bus */
+          if (bus_type == G_BUS_TYPE_SYSTEM)
+            {
+              g_autoptr(GVariant) update_driver_docs_array = NULL;
+              g_autoptr(GError) local_error = NULL;
+              GDBusConnection *connection =
+                g_dbus_proxy_get_connection (G_DBUS_PROXY (sysroot_proxy));
+              if (!get_update_driver_doc (connection, update_driver_sd_unit, &update_driver_docs_array,
+                                          cancellable, &local_error))
+                {
+                  g_printerr ("%s", local_error->message);
+                }
+              else if (update_driver_docs_array)
+                {
+                  gsize docs_len;
+                  g_autofree const char **update_driver_docs =
+                    g_variant_get_strv (update_driver_docs_array, &docs_len);
+                  if (docs_len > 0)
+                    {
+                      g_print ("; see the documentation for that software at ");
```

Comment:
I'd prefer to break this up into multiple lines instead of using `;` to join them.  The URL could be long.